### ISSUE TITLE
(Simplified?) Actor: Invoke Script Replacement

### ISSUE THREAD

**Issue by SetsuneW** [chrismaltby/gb-studio#1138]
**Is your feature request related to a problem? Please describe.**
Being able to programmatically activate another actor's script is an incredibly useful feature. It was available in GB Studio v2, and I used it to create a custom description system, sequence events, run timers, and on the whole encapsulate instructions so it was much easier to keep code clean.

Unfortunately when trying to convert over the GB Studio v3, I discovered Invoke Actor was depreciated. It's supposed to sub in an alternate method to allow for it, but the project refuses to build with the instruction in place at all. (So this is a bit of a bug report too.)

Based on a discussion I had on the Discord, this is apparently due to the complete rewrite of the scripting back end, and how actor scripts are run now. They can't call another actor because of their intentionally limited scope, so to speak, and to avoid some of the memory issues and crashes v2 ran into. But I'm hoping there's still room for a limited implementation?

**Describe the solution you'd like**
Barring a full reimplementation, I'd like to know if it would be possible to include a simplified alternative that might still work. Primarily, all it has to do is "fake an A press" on the Actor. I'm unclear how much information needs to be passed to the actor to make that A press work (such as Player position).

Without knowing the full structure of how actor scripts work in GB Studio, for the sake of memory concerns, etc, maybe the command becomes "End Script and Invoke Actor". The script writes a value to a system variable that would normally be zero and ends the current script. On return from an actor script, GB Studio checks for a value in that variable, and if it's set to an actor ID (+ whatever information it may need like interaction direction), it acts just like the player immediately spoke to that actor following the last and runs it accordingly. This might be another actor, or the same actor if you need to "reset" their script.

This doesn't give that "nested" functionality of resuming where the original script left off, but it still handles a lot of use cases I designed around personally. In most cases I had no intention of returning back to the original actor, and if I did, I'd simply Invoke them in turn.

**Describe alternatives you've considered**
I could hypothetically create a script myself, but I'm not sure how I'd activate it to check automatically following an interaction, and I still don't have the functionality to Invoke the Actor, to "Fake that A Press", so to speak, so it's back to square one. This might be possible with a plug-in or modification to the core, but that seems a bit beyond me at the moment. And I feel like this would be a functional tool for the software itself.

**Comment by maxoakland** [comment]
I agree that this is an important and necessary feature that should be reimplemented
